src/test/java/com/google/api/generator/engine/ast/CastExprTest.java

```diff
@@ -73,6 +73,49 @@ public void validCastExpr_basicPrimitiveWidening() {
 
   }
 
+  @Test
+  public void validCastExpr_castPrimitiveToBoxedType() {
+    PrimitiveValue intValue = PrimitiveValue.builder().setValue("3").setType(TypeNode.INT).build();
+    ValueExpr valueExpr = ValueExpr.withValue(intValue);
+    CastExpr expr = CastExpr.builder().setType(TypeNode.INT_OBJECT).setExpr(valueExpr).build();
+  }
+
+  @Test
+  public void validCastExpr_castBoxedTypeToPrimitive() {
+    Variable variable = Variable.builder().setName("x").setType(TypeNode.DOUBLE_OBJECT).build();
+    VariableExpr variableExpr = VariableExpr.builder().setVariable(variable).build();
+    CastExpr expr = CastExpr.builder().setType(TypeNode.DOUBLE).setExpr(variableExpr).build();
+  }
+
+  @Test
```

Comment:
Great to see that these edge cases were considered. In this case, both `nestedPrimitiveBoxedType` tests are already covered by the tests for casting primitives to objects and vice versa, so we can remove them.

Reply:
done. Thanks!